### ISSUE TITLE
Setup uses future data

### ISSUE THREAD

**Issue by ghcmartins** [pycaret/pycaret#219]
Hi Pycaret team!

I´ve made a test today to see if the setup models was using future data and it seems to do be doing it.
The test:
-csv file with 5000 daily data from SPY ETF(total base)
-csv file with 2778 oldest days  from SPY ETF(training base)

1st Run:
-Using the total base file (5000) and setting train_size = 0.5 (2500 days for train)
-Results on the evaluate model  (0.61 AUC)
-Exporting results to a csv with % accuracy from predict_model output when checked

2nd Run:
-Using the training base (2778 days_) setting train_size = 0.9 (2500 days for train)
-Results on tehe valuate model when predicting the rest of the base (2222 days on test/hold-out), were (AUC 0.5)

There´s clearly something on the preprocessing causing this, it should be the same results (session_id was the same in both runs, same setup config). If someone could test this with stocks and see if it can be fixed, would be great!

Best,

**Comment by pycaret** [comment]
@ghcmartins Ho,

Thanks for reporting. Are you trying to do time series? Maybe if you share your files and script that will help to understand what you are saying. 

Thanks

**Comment by ghcmartins** [comment]
Sure! I´ll attach it here in the thread.

**Comment by pycaret** [comment]
@ghcmartins 

I think I got your point. The reason this is happening is because PyCaret is shuffling the data in train_test_split. At the moment there is no option to set that to False but in next release there would be a parameter in setup named 'split_shuffle' that will control the ability of shuffle. 

The next release is expected to be out in next 30 days. 

I will keep the thread open until then. Once release is out, I will close this issue.

**Comment by ghcmartins** [comment]
Thank you very much, it´s important to avoid "too good" results, it´s what I was having 60-80% accuracy on time series (I was feeling like king of the quants!LOL)

**Comment by ghcmartins** [comment]
Also the gold example gives results "too good to be true", so I´d suggest to add this note on the notebook also.

**Comment by pycaret** [comment]
@ghcmartins Thanks.

I am tagging this as an enhancement as the feature you are looking for is to disable shuffle in split which is coming in next release. I will close this issue when next release is out.

Thanks.

**Comment by pycaret** [comment]
shuffle parameter now available in **pycaret-nightly**

**Comment by volkanakcora** [comment]
I am getting that error when I execute: setup()

Setting a random_state has no effect since shuffle is False. You should leave random_state to its default (None), or set shuffle=True.

**Comment by Moktar13** [comment]
Did you end up finding a fix for this issue?

**Comment by agusxpro** [comment]
I put optional argument:
fold_shuffle = True
inside the setup function, it worked fine.
